src/k8s/pkg/k8sd/app/hooks_remove.go

```diff
@@ -122,5 +113,21 @@ func (a *App) onPreRemove(ctx context.Context, s state.State, force bool) (rerr
 		}
 	}
 
+	if _, ok := cfg.Annotations[apiv1.AnnotationSkipCleanupKubernetesNodeOnRemove]; ok {
+		// Explicitly skip removing the node from Kubernetes.
+		log.Info("Skipping Kubernetes control-plane node removal")
```

Comment:
maybe make `log.WithValues("node", s.Name())` earlier when you create the logger